groups/csa/complexity_check.cpp

```diff
@@ -0,0 +1,143 @@
+        findAll(stmt(catchStmt().bind("catch"))),
+        findAll(stmt(conditionalOperator().bind("?operator"))),
+        findAll(stmt(ifStmt(hasCondition(findAll(binaryOperator(hasOperatorName("&&")).bind("condOp")))))),
+        findAll(stmt(ifStmt(hasCondition(findAll(binaryOperator(hasOperatorName("||")).bind("condOp"))))))
+        )))).bind("methodDecl"));
+
+    return matcher; 
+}
+
+
+void report::operator()()
+{
+    MatchFinder finder;
+    csabase::OnMatch<report, &report::calculateComplex> m(this);
+    finder.addDynamicMatcher(dyn_matchComplexity(), &m);
+    finder.match(*d_analyser.context()->getTranslationUnitDecl(), *d_analyser.context());
+    generateReport();
+}
+
+// Generates report if any of the methods exceeded the verified threshold
+void report::generateReport(){
+    // Get the complexity threshold
+    data& d = a.attachment<data>();
+    std::string tag = "CX01";
+    std::string message = "Complexity Exceeded the maximum threshold ! Threshold verified: " 
+    + std::to_string(d.d_max_complexity) + " Found: ";
+    //location of method
+    SourceLocation loc; 
+    
+    for(auto curMethod : complexDict){
+        if (curMethod.second > d.d_max_complexity){
+            loc = curMethod.first->getLocStart();
+            std::string val = std::to_string(curMethod.second);
+            a.report(loc, check_name, tag, message + val);
+        }
+        // Add to total File complexity
+        complexity += curMethod.second;
+    }
+    
+}
+
+
+void report::calculateComplex(BoundNodes const & nodes)
+{
+    FunctionDecl const* curMethod = nodes.getNodeAs<FunctionDecl>("methodDecl");
+    IfStmt const * curIf = nodes.getNodeAs<IfStmt>("if");
+    SwitchCase const * curSwitch = nodes.getNodeAs<SwitchCase>("switch");
+    WhileStmt const * curWhile = nodes.getNodeAs<WhileStmt>("while");
+    ForStmt const * curFor = nodes.getNodeAs<ForStmt>("for");
+    CXXTryStmt const * curTry = nodes.getNodeAs<CXXTryStmt>("try");
+    CXXCatchStmt const * curCatch = nodes.getNodeAs<CXXCatchStmt>("catch");
+    BinaryOperator const * curBin = nodes.getNodeAs<BinaryOperator>("condOp");
+    ConditionalOperator const * curCond = nodes.getNodeAs<ConditionalOperator>("?operator");
+
+    SourceManager &sm = a.manager();
+    const auto& fId = sm.getMainFileID();
+    const auto& fileId = sm.getFileID(curMethod->getLocation());
+
+    // If matched method is not in the same .cpp file i.e:includes
+    if (fId == fileId){
+        if (curWhile != nullptr || curSwitch != nullptr || curFor != nullptr || curIf != nullptr
+            || curTry != nullptr || curCatch != nullptr || curBin != nullptr || curCond != nullptr){
+            if (complexDict.find(curMethod) != complexDict.end()){
+                complexDict[curMethod] = complexDict[curMethod] + 1;    
+            }else{
+                // 2: a basic 1 for each method + the first occurence
+                complexDict[curMethod] = 2;
+            }
+        }
+    }
```

Comment:
Looks like you are completely ignoring function that do not have any type of conditional, I do not see how a simple function will have a complexity of 1